tools/istio-iptables/pkg/cmd/run.go

```diff
@@ -861,17 +861,36 @@ func (iptConfigurator *IptablesConfigurator) executeIptablesRestoreCommand(isIpv
 
 func (iptConfigurator *IptablesConfigurator) executeCommands() {
 	if iptConfigurator.cfg.RestoreFormat {
-		// Execute iptables-restore
-		err := iptConfigurator.executeIptablesRestoreCommand(true)
-		if err != nil {
-			log.Errorf("Failed to execute iptables-restore command: %v", err)
-			os.Exit(1)
+
+		// Get the standard output of the iptables-save command
+		output, _ := iptConfigurator.ext.CmdOutput(constants.IPTABLESSAVE, "-t", constants.NAT)
+
+		// Skip execution of iptables-restore if it has already run before
+		// This ensures safe re-execution of init containers
+		if strings.Contains(output, "ISTIO") {
+			log.Infof("ISTIO chain(s) already exist in the %s table of the %s", constants.NAT, constants.IPTABLES)
+		} else {
+			// Execute iptables-restore
+			err := iptConfigurator.executeIptablesRestoreCommand(true)
+			if err != nil {
+				log.Errorf("Failed to execute iptables-restore command: %v", err)
+				os.Exit(1)
+			}
 		}
-		// Execute ip6tables-restore
-		err = iptConfigurator.executeIptablesRestoreCommand(false)
-		if err != nil {
-			log.Errorf("Failed to execute iptables-restore command: %v", err)
-			os.Exit(1)
+
+		// Get the standard output of the ip6tables-save command
+		output, _ = iptConfigurator.ext.CmdOutput(constants.IP6TABLESSAVE, "-t", constants.NAT)
+
+		// Skip execution of ip6tables-restore for the same reason as IPv4 above
+		if strings.Contains(output, "ISTIO") && iptConfigurator.cfg.EnableInboundIPv6 {
+			log.Infof("ISTIO chain(s) already exist in the %s table of the %s", constants.NAT, constants.IP6TABLES)
+		} else {
+			// Execute ip6tables-restore
+			err := iptConfigurator.executeIptablesRestoreCommand(false)
+			if err != nil {
+				log.Errorf("Failed to execute iptables-restore command: %v", err)
+				os.Exit(1)
+			}
 		}
 	} else {
```

Comment:
shouldn't rerun enters the else clause?

Reply:
No, rerun doesn't enter the else clause. "_restore-format_" config is the condition that is being checked on line 863 and this is set to **true** by default (`viper.SetDefault(constants.RestoreFormat, true)`). Check the flag here: https://istio.io/latest/docs/reference/commands/pilot-agent/#pilot-agent-istio-iptables

So, even during rerun it would come inside the if clause. Ideally, my changes would be needed in the else clause here as well, if for whatever reason _restore-format_ is set to false.

Reply:
got it